android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -578,6 +637,62 @@ constructor(
         }
       }
     }
+
+    @JvmOverloads
+    fun updateResource(
+      resource: Resource?,
+      path: String?,
+      value: Any?,
+      purgeAffectedResources: Boolean = false,
+      createLocalChangeEntitiesAfterPurge: Boolean = true,
+    ) {
+      if (resource == null || path.isNullOrEmpty()) return
+
+      val jsonParse = JsonPath.using(conf).parse(resource.encodeResourceToString())
+
+      val updatedResourceDocument =
+        try {
+          jsonParse.apply {
+            // Expression stars with '$' (JSONPath) or ResourceType like in FHIRPath
+            if (path.startsWith("\$") && value != null) {
+              set(path, value)
+            }
+            if (
+              path.startsWith(
+                resource.resourceType.name,
+                ignoreCase = true,
+              ) && value != null
+            ) {
+              set(
+                path.replace(resource.resourceType.name, "\$"),
+                value,
+              )
+            }
+
+            if (resource.id.startsWith("#")) {
+              val idPath = "\$.id"
+              set(idPath, resource.id.replace("#", ""))
+            }
+          }
+        } catch (e: PathNotFoundException) {
+          Timber.e(e, "Path $path not found")
+          jsonParse
+        }
+
+      val resourceDefinition: Class<out IBaseResource>? =
+        FhirContext.forR4Cached().getResourceDefinition(resource).implementingClass
+
+      val updatedResource =
+        parser.parseResource(resourceDefinition, updatedResourceDocument.jsonString())
+      CoroutineScope(dispatcherProvider.io()).launch {
+        if (purgeAffectedResources) {
+          defaultRepository.purge(updatedResource as Resource, forcePurge = true)
+        }
+        if (createLocalChangeEntitiesAfterPurge) {
+          defaultRepository.addOrUpdate(resource = updatedResource as Resource)
+        } else defaultRepository.createRemote(resource = *arrayOf(updatedResource as Resource))
```

Comment:
```suggestion
        } else defaultRepository.createRemote(resource = arrayOf(updatedResource as Resource))
```

The spread operator(*) is redunandant